server/createBinaries.js

```diff
@@ -41,97 +41,129 @@ createBinaries = function() {
   var finalDir = path.join(workingDir, "final");
   mkdirp(finalDir);
 
-
-  var electronSettings = Meteor.settings.electron || {};
-  var appVersion = electronSettings.version;
-  var appName = electronSettings.name;
-
-  [
-    "autoUpdater.js",
-    "main.js",
-    "menu.js",
-    "package.json",
-    "preload.js",
-    "proxyWindowEvents.js"
-  ].forEach(function(filename) {
-    var fileContents = Assets.getText(path.join("app", filename));
-
-    // Replace parameters in `package.json`.
-    if (filename === "package.json") {
-      var packageJSON = JSON.parse(fileContents);
-      if (appVersion) packageJSON.version = appVersion;
-      if (appName) {
-        packageJSON.name = appName.toLowerCase().replace(/\s/g, '-');
-        packageJSON.productName = appName;
-      }
-      fileContents = JSON.stringify(packageJSON);
-    }
-
-    writeFile(path.join(appDir, filename), fileContents);
-  });
-
-  //TODO be smarter about caching this..
-  exec("npm install", {cwd: appDir});
-
-  var settings = _.defaults({}, electronSettings, {
-    rootUrl: process.env.ROOT_URL
-  });
-
-  var signingIdentity = electronSettings.sign;
-  if (canServeUpdates()) {
-    // Enable the auto-updater if possible.
-    if ((platform === 'darwin') && !signingIdentity) {
-      // If the app isn't signed and we try to use the auto-updater, it will
-      // throw an exception.
-      console.error('Developer ID signing identity is missing: remote updates will not work.');
-    } else {
-      settings.updateFeedUrl = settings.rootUrl + UPDATE_FEED_PATH;
-    }
+  return {
+    working: workingDir,
+    binary: binaryDir,
```

Comment:
The binary dir could be shared between platforms. App will be slightly different between each platform (sightly different electronSettings based on signing, icons etc.) It would be better to share more of the app between builds to prevent double running of slow processes like `npm install`, but it will require a little thought to do it. I created a separate issue https://github.com/rissem/meteor-electron/issues/35 to track this.